pandas/tests/groupby/test_function.py

```diff
@@ -1015,6 +1025,45 @@ def test_nunique_with_timegrouper():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "data, expected",
+    [
+        (
+            DataFrame(
+                {
+                    "key": ["x", "x", "x", "x", "x"],
+                    "data": [
+                        Timestamp("2019-01-01 00:00:00"),
+                        NaT,
+                        Timestamp("2019-01-01 00:00:00"),
+                        NaT,
+                        Timestamp("2019-01-01 00:00:00"),
+                    ],
+                }
+            ),
+            1,
+        ),
+        (
+            DataFrame(
+                {
+                    "key": ["x", "x", "x", "x", "x"],
+                    "data": [
+                        dt.date(2019, 1, 1),
+                        NaT,
+                        dt.date(2019, 1, 1),
+                        NaT,
+                        dt.date(2019, 1, 1),
+                    ],
+                }
+            ),
+            1,
+        ),
+    ],
+)
+def test_nunique_with_NaT(data, expected):
+    assert data.groupby(["key"])["data"].nunique()[0] == expected
```

Comment:
Can you add the issue number as a comment?

Also, maybe a little more readable to explicitly define a `result`, then `assert`:
```python
result = data.groupby(["key"])["data"].nunique()[0]
assert result == expected
```

Reply:
Sure, thanks for your feedback!